src/types/boolobject.rs

```diff
@@ -14,12 +14,33 @@ pub struct PyBool(PyAny);
 pyobject_native_type!(PyBool, ffi::PyObject, pyobject_native_static_type_object!(ffi::PyBool_Type), #checkfunction=ffi::PyBool_Check);
 
 impl PyBool {
-    /// Depending on `val`, returns `true` or `false`.
+    /// Deprecated form of [`PyBool::new_bound`]
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`PyBool::new` will be replaced by `PyBool::new_bound` in a future PyO3 version"
+        )
+    )]
     #[inline]
     pub fn new(py: Python<'_>, val: bool) -> &PyBool {
         unsafe { py.from_borrowed_ptr(if val { ffi::Py_True() } else { ffi::Py_False() }) }
     }
 
+    /// Depending on `val`, returns `true` or `false`.
+    ///
+    /// # Note
+    /// This returns a [`Borrowed`] reference to one of Pythons `True` or
+    /// `False` singletons
+    #[inline]
+    pub fn new_bound(py: Python<'_>, val: bool) -> Borrowed<'_, '_, Self> {
```

Comment:
Sorry for being late to the party: If these are singletons, shouldn't the first lifetime even be `'static`, i.e.

```rust
pub fn new_bound(py: Python<'py>, val: bool) -> Borrowed<'static, 'py, Self>;
```

or are they still bound to the interpreter?

Reply:
I wondered about this, I think while static is possibly fine I preferred having this limited to the interpreter. Just in case this becomes relevant for things like subinterpreters in the future.

Reply:
I guess we can always expand this later if users have a need.

Reply:
I thought about this too, but it seemed safer to go with this option now, and relax it in the future if possible.